beam/www/beam/components/ControlButtons.vue

```diff
@@ -0,0 +1,61 @@
+<template>
+	<div class="control-buttons">
+		<button @click="onCreate" :disabled="docstatus !== -1">Save</button>
+		<button @click="onSubmit" :disabled="docstatus !== 0">Submit</button>
+		<button @click="onCancel" :disabled="docstatus !== 1">Cancel</button>
+	</div>
+</template>
+
+<script setup lang="ts">
```

Comment:
To avoid having to use `any` everywhere in the component and add additional typing in the IDE, you could add a [generic type](https://vuejs.org/api/sfc-script-setup#generics) here. Something like:

```suggestion
<script setup lang="ts" generic="T extends ParentDoctype">
```

That way, your `props` can change to:

```ts
const props = defineProps<{
	onCreate: () => Promise<{ data: T }>
	onSubmit: () => Promise<{ data: T; exception: string; response: Response }>
	onCancel: () => Promise<{ data: T; exception: string; response: Response }>
}>()
```

And you automatically get the following in the IDE:

![image](https://github.com/user-attachments/assets/be70507c-7ba5-41f5-b1c1-11927115fed7)